File: posts-service/src/main/java/org/example/postsservice/controllers/HealthCheckController.java
package org.example.postsservice.controllers;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/posts/health")
@CrossOrigin(origins = "*")
public class HealthCheckController {
    @GetMapping("")
    public ResponseEntity<String> health() {
        return ResponseEntity.ok("OK");
    }
}

File: follow-service/src/main/kotlin/org/example/followservice/controllers/HealthCheckController.kt
package org.example.followservice.controllers

import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/follow/health")
@CrossOrigin(origins = ["*"])
class HealthCheckController {
    @GetMapping("")
    fun healthCheck(): String {
        return "Follow Service is up and running"
    }
}